e2e/system_test.go

```diff
@@ -0,0 +1,201 @@
+			big.NewInt(0),
+		},
+	}
+
+	stakingAddress := types.StringToAddress("1001")
+
+	testTable := []struct {
+		name          string
+		staker        types.Address
+		stakeAmount   *big.Int
+		shouldSucceed bool
+	}{
+		{
+			"Valid stake",
+			validAccounts[0].address,
+			framework.EthToWei(10),
+			true,
+		},
+		{
+			"Invalid stake",
+			validAccounts[1].address,
+			framework.EthToWei(100),
+			false,
+		},
+	}
+
+	srvs := framework.NewTestServers(t, 1, func(config *framework.TestServerConfig) {
+		config.SetConsensus(framework.ConsensusDev)
+		config.SetSeal(true)
+
+		for _, acc := range validAccounts {
+			config.Premine(acc.address, acc.balance)
+		}
+	})
+	srv := srvs[0]
+
+	rpcClient := srv.JSONRPC()
+	for indx, testCase := range testTable {
+		t.Run(testCase.name, func(t *testing.T) {
+			// Fetch the staker balance before sending the transaction
+			accountBalance, err := rpcClient.Eth().GetBalance(
+				web3.Address(testCase.staker),
+				web3.Latest,
+			)
+			assert.NoError(t, err)
+
+			var out string
+			if callErr := rpcClient.Call(
+				"stake_getStakedBalance",
+				&out,
+				web3.Address(testCase.staker),
+				web3.Latest.String(),
+			); callErr != nil && testCase.shouldSucceed {
+				t.Fatalf("Unable to fetch staked balance")
+			}
+
+			stakedBalance, ok := new(big.Int).SetString(out[2:], 16)
+			if !ok {
+				t.Fatalf("Unable to convert staked balance response to big.Int")
+			}
+
+			// Set the preSend balances
+			previousAccountBalance, _ := big.NewInt(0).SetString(accountBalance.String(), 10)
+			previousStakedBalance, _ := big.NewInt(0).SetString(stakedBalance.String(), 10)
+
+			// Create the transaction
+			txnObject := &types.Transaction{
+				From:     testCase.staker,
+				To:       &stakingAddress,
+				GasPrice: big.NewInt(1048576),
+				Gas:      1000000,
+				Value:    testCase.stakeAmount,
+			}
+
+			signedTxn, err := signer.SignTx(txnObject, addressKeyPairs[indx].privateKey)
+			assert.NoError(t, err)
+
+			data := signedTxn.MarshalRLP()
+
+			fee := big.NewInt(0)
+
+			// Do the transfer
+			txnHash, err := rpcClient.Eth().SendRawTransaction(data)
+			assert.NoError(t, err)
+			assert.IsTypef(t, web3.Hash{}, txnHash, "Return type mismatch")
+
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+
+			receipt, err := srv.WaitForReceipt(ctx, txnHash)
+
+			assert.NoError(t, err)
+			assert.NotNil(t, receipt)
+
+			if testCase.shouldSucceed {
+				fee = new(big.Int).Mul(
+					big.NewInt(int64(receipt.GasUsed)),
+					big.NewInt(txnObject.GasPrice.Int64()),
+				)
+			}
+
+			// Fetch the balance after sending
+			accountBalance, err = rpcClient.Eth().GetBalance(
+				web3.Address(testCase.staker),
+				web3.Latest,
+			)
+			assert.NoError(t, err)
+
+			if callErr := rpcClient.Call(
+				"stake_getStakedBalance",
+				&out,
+				web3.Address(testCase.staker),
+				web3.Latest.String(),
+			); callErr != nil && testCase.shouldSucceed {
```

Comment:
It should return the result (0) even if the account doesn't exist, so no need for the second part of the condition, right?
```suggestion
			); callErr != nil {
```

Reply:
Fixed 👍